adding updatetestcase method

eitzenbe opened the pull request:
added retrieval of status and version on create test case (in the
util.getTestcase method)

This also includes the functionality of the version pull request (I assume) from dida43

kinow commented on src/main/java/br/eti/kinoshita/testlinkjavaapi/TestCaseService.java:
```diff
@@ -114,6 +116,18 @@ protected TestCase createTestCase(String testCaseName, Integer testSuiteId, Inte
         return testCase;
     }
 
+    public void updateTestCase(TestCase tc) throws TestLinkAPIException {
```
The PHP method returns a mixed array.

On success:

    array('status_ok' => true, 'msg' => 'ok', 'operation' => __FUNCTION__);

And on error returns the default errors object.

kinow commented on .gitignore:
```diff
@@ -1,6 +1,5 @@
+/target/
+/.settings/
+/.classpath
+/.project
```
Sometimes the TAP integration, while playing with it, creates the test-output folder. Any reason to remove it here? We can include the /bin/ folder without removing it I think?

kinow commented on src/main/java/br/eti/kinoshita/testlinkjavaapi/model/TestCase.java:
```diff
@@ -140,7 +144,8 @@ public Integer getParentId() {
 
     /**
-     * @param parentId the parentId to set
+     * @param parentId
```
Always good to avoid formatting changes. See pull request #37 :cry:

kinow commented:
HI @eitzenbe, few nit-picks like formatting, but don't mind about these. If you could fix just the return of that method, to return the array. Like in updateTestCaseCustomFieldDesignValue.

The shoemakers children go barefoot, as you may notice looking at assignTestCaseExecutionTask, where I forgot to add that.

In case you don't have the time, we can merge it on the next development cycle, and I will take care to fix the return statement.

Excellent work! Fixed a bunch of other issues, and merged way more pull requests than I thought I would be able to today. 

Release scheduled for Sunday noon, New Zealand time.

B

eitzenbe commented:
Hi  B ;)
I am traveling today so if you could add the return that would be awesome. Alternatively if you could wait till Sunday evening for release:I am home tonight and then could make the mods. Any insight on the execution time documentation issue when uploading results?
All the best
T
Am 08.04.2016 09:14 schrieb Thomas Eitzenberger <thomas@eitzen.at>:Hi  B ;)
I am traveling today so if you could add the return that would be awesome. Alternatively if you could wait till Sunday evening for release:I am home tonight and then could make the mods. Any insight on the execution time documentation issue when uploading results?
All the best
T
Am 08.04.2016 07:23 schrieb "Bruno P. Kinoshita" <notifications@github.com>:HI @eitzenbe, few nit-picks like formatting, but don't mind about these. If you could fix just the return of that method, to return the array. Like in updateTestCaseCustomFieldDesignValue.

The shoemakers children go barefoot, as you may notice looking at assignTestCaseExecutionTask, where I forgot to add that.

In case you don't have the time, we can merge it on the next development cycle, and I will take care to fix the return statement.

Excellent work! Fixed a bunch of other issues, and merged way more pull requests than I thought I would be able to today. 

Release scheduled for Sunday noon, New Zealand time.

B

—You are receiving this because you were mentioned.Reply to this email directly or view it on GitHub

eitzenbe commented:
added the return feature and javadoced the API method...

you have a gogo AFTER testing it, my testlink instance was shutdown by
some admin and I need to check on monday whats wrong...

br
Thomas

Am 2016-04-08 um 07:23 schrieb Bruno P. Kinoshita:
>
> HI @eitzenbe <https://github.com/eitzenbe>, few nit-picks like
> formatting, but don't mind about these. If you could fix just the
> return of that method, to return the array. Like in
> updateTestCaseCustomFieldDesignValue.
>
> The shoemakers children go barefoot, as you may notice looking at
> assignTestCaseExecutionTask, where I forgot to add that.
>
> In case you don't have the time, we can merge it on the next
> development cycle, and I will take care to fix the return statement.
>
> Excellent work! Fixed a bunch of other issues, and merged way more
> pull requests than I thought I would be able to today.
>
> Release scheduled for Sunday noon, New Zealand time.
>
> B
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kinow/testlink-java-api/pull/67#issuecomment-207219982>
>

-- 

Crimes against Nature are Crimes
against Peace and Humanity.
Protect our natural home.
https://www.endecocide.org

kinow commented:
You rock @eitzenbe !!! Merging.

kinow commented:
Included in changes.xml for the next release (see 9aeae835984ffcc15117172343a6366206cfb754). The last issue that I will try to take a look at before cutting the release is the pull request #39.

Once that's done, will push to Maven Central. Thanks again Thomas!

eitzenbe commented:
Grande MISERA!

I just tested the status feature and traced the xmlrpc api code on
server side and found out , that the status attribute of the test case
is NOT supported there :(

it doesnt do any harm, but it also does NOT update the test case status
in the system...

Sorry
Thomas

Am 2016-04-09 um 03:37 schrieb Bruno P. Kinoshita:
>
> Included in changes.xml for the next release (see 9aeae83
> <https://github.com/kinow/testlink-java-api/commit/9aeae835984ffcc15117172343a6366206cfb754>).
> The last issue that I will try to take a look at before cutting the
> release is the pull request #39
> <https://github.com/kinow/testlink-java-api/pull/39>.
>
> Once that's done, will push to Maven Central. Thanks again Thomas!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kinow/testlink-java-api/pull/67#issuecomment-207678193>
>

-- 

Crimes against Nature are Crimes
against Peace and Humanity.
Protect our natural home.
https://www.endecocide.org

kinow commented:
Fixed I think https://github.com/kinow/testlink-java-api/commit/eb9ac19efb822f141aa71fbe3495f428a889ae55#diff-f90c6e727736dc94227cff1e470bba5fR512

Re-packaging Jenkins Plug-in with the snapshot. Will git it a go, and if it works, releasing in a few minutes.

eitzenbe commented:
http://mantis.testlink.org/view.php?id=7497
is the ticket for the xmlrpc ;)

Am 2016-04-10 um 00:07 schrieb Bruno P. Kinoshita:
>
> Fixed I think eb9ac19#diff-f90c6e727736dc94227cff1e470bba5fR512
> <https://github.com/kinow/testlink-java-api/commit/eb9ac19efb822f141aa71fbe3495f428a889ae55#diff-f90c6e727736dc94227cff1e470bba5fR512>
>
> Re-packaging Jenkins Plug-in with the snapshot. Will git it a go, and
> if it works, releasing in a few minutes.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kinow/testlink-java-api/pull/67#issuecomment-207863087>
>

-- 

Crimes against Nature are Crimes
against Peace and Humanity.
Protect our natural home.
https://www.endecocide.org

kinow commented:
All done. Preparing release! Available in Maven central in the next hours.

eitzenbe commented:
HOLD THE HORSES

Quick addon in TastCaseService.createTestCase PLEASE
also populate the version number form the response into the test case
object being returned!

Thanx Thomas

Am 2016-04-10 um 02:14 schrieb Bruno P. Kinoshita:
>
> All done. Preparing release! Available in Maven central in the next hours.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kinow/testlink-java-api/pull/67#issuecomment-207890032>
>

-- 

Crimes against Nature are Crimes
against Peace and Humanity.
Protect our natural home.
https://www.endecocide.org
